x/params/store/store.go

```
@@ -0,0 +1,176 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Individual parameter store for each keeper
+// Transient store persists for a block, so we use it for
+// recording whether the parameter has been changed or not
+type Store struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey // []byte -> []byte, stores parameter
+	tkey sdk.StoreKey // []byte -> bool, stores parameter change
+
+	space []byte
+}
+
+// NewStore constructs a store with namestore
+func NewStore(cdc *codec.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space),
+	}
+}
+
+// Returns a KVStore identical with ctx,TransientStore(s.key).Prefix()
+func (s Store) kvStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.KVStore(s.key).Prefix(append(s.space, '/'))
+}
+
+// Returns a KVStore identical with ctx.TransientStore(s.tkey).Prefix()
+func (s Store) transientStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.TransientStore(s.tkey).Prefix(append(s.space, '/'))
+}
+
+// Get parameter from store
+func (s Store) Get(ctx sdk.Context, key []byte, ptr interface{}) {
+	store := s.kvStore(ctx)
+	bz := store.Get(key)
+	err := s.cdc.UnmarshalJSON(bz, ptr)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// GetIfExists do not modify ptr if the stored parameter is nil
+func (s Store) GetIfExists(ctx sdk.Context, key []byte, ptr interface{}) {
```

Why not just return the err if it doesn't exist?